### ISSUE TITLE
Not informative INTERNAL_SERVER_ERROR when mutation has invalid input as variable

### ISSUE THREAD

**Issue by ph55** [apollographql/apollo-server#2204]
If mutation request made with input variable and it has invalid data - error not informative.
Example:

* Good. Login mutation with regular input
```gql
mutation {
  login(input: {
    username: "user",
    password: 1
  })
}
```
Response: 
```json
{
  "errors": [
    {
      "message": "Expected type String!, found 1.",
      "locations": [
        {
          "line": 4,
          "column": 13
        }
      ],
      "extensions": {
        "code": "GRAPHQL_VALIDATION_FAILED"
      }
    }
  ]
}
```

---

* Bad. Login mutation with **variable** input
```gql
mutation login($input: LoginInput!) {
  login(input: $input)
}
```
variables:
```json
{
  "input": {
    "username": "user",
    "password": 1
   }
}
```

Response: 
```json
{
  "errors": [
    {
      "message": "Internal server error",
      "locations": [
        {
          "line": 1,
          "column": 16
        }
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

As you can see there is zero information regarding error. Also it should be `GRAPHQL_VALIDATION_FAILED` instead of `INTERNAL_SERVER_ERROR`.
Only if with `debug` on you can see the error:

```json
{
  "errors": [
    {
      "message": "Internal server error",
      "locations": [
        {
          "line": 1,
          "column": 16
        }
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "TypeError: String cannot represent a non string value: 0",
            "........."
          ]
        }
      }
    }
  ]
}
```

Probably related to https://github.com/apollographql/apollo-server/issues/1410
Still opening this one since it have a bit more information.

This is a big issue since normal API couldn't be built without error info.

**Comment by kfrajtak** [comment]
I am afraid this does not work in 2.6.7.

**Comment by glasser** [comment]
Duplicate of #3498

**Comment by glasser** [comment]
I've released a prerelease with this fix, version `2.23.0-alpha.0`. Try out the alpha and see if it works for you! Please provide any feedback in #5094.

**Comment by glasser** [comment]
(as a workaround you can also switch your setup to using payloadFormatVersion 1.0)

**Comment by glasser** [comment]
This is released in Apollo Server 2.23.0.
